GOVERNANCE.md

```diff
@@ -0,0 +1,251 @@
+# Governance thin-edge.io
+
+This documents servers as a guiding document to ease technical decision making in the project as a whole.
+
+Some of its goals are:
+
+- Making sure that the different stakeholders are represented with their requirements
+- Making sure that contributors know how decisions are made and who they can address concerns/questions to
+- Making sure that team members know what is required of them for the project
+
+
+## The project vision
+
+In the [vision.md](./vision.md) the goal of the project is encoded.
+It specifies the different functional requirements and non-functional requirements of the project as a whole.
+All the stakeholders must agree on the vision upon joining.
+
+- The current stakeholders are: Software AG, and IFM
+
+Changes to the vision need to be done in a plenum of stakeholders, this makes sure that everyone is aware and agrees to the evolution of the project.
+
+
+
+## Team Structure
+
+* Teams are either single contributors or a group of them
+* They are responsible for a specific part of project, and as such are the maintainers of those parts
+    * This includes: Reviewing pull requests, triaging issues assigned to them, and general maintenance
+* Each team is assigned a single or multiple subtrees of the project, as defined by the project structure
+* The teams follow the project hierarchy for decisions, i.e. higher decisions take precedence
+
+The important bits:
+
+At the top, each invested party has their representative, who together with the others in their group have to come to conclusions together. These should be based on the following factors:
+
+- The signed off vision of the whole project
+- Feasability, accounting for future technical and cultural shifts
+- The health of the project going forward
+
+Overall, these 'project leads' should not be beholden to their respective companies, and instead to the health of the project and its community.
+
+Underneath the leads are the teams or individuals, that each are allowed to  make similar decisions about the project *in the area they have been delegated*.
+
+
+- Some decisions are global in the project, like the core/API
+    - Team members should try and make sure to explore possible options together before calling for a vote. This mostly includes uncontroversial changes that are low impact or have already been agreed upon
+    - If this is not possible, a vote needs to be held: TODO
```

Comment:
```suggestion
    - If an exclusive choice has to be made (where it is not possible to entertain two conflicting approaches), and no clear side is more advantageous to pick, a vote should be held with the majority opinion being the final decision.
    - This does not preclude that once the tradeoff changes, this might be reverted or otherwise adapted.
```

Reply:
Updated